File: app/src/main/java/com/example/movieslisttask/view/ui/activities/MovieInfoActivity.java
package com.example.movieslisttask.view.ui.activities;

import android.content.Intent;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;

import com.example.movieslisttask.R;
import com.example.movieslisttask.databinding.ActivityMovieInfoBinding;
import com.example.movieslisttask.model.Movie;

/**
 * Created by alexkorolov on 2020-01-17.
 */
public class MovieInfoActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_movie_info);
        ActivityMovieInfoBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_movie_info);

        Intent intent = getIntent();
        if (intent.hasExtra("movie")) {
            Movie movie = getIntent().getParcelableExtra("movie");
            binding.setMovie(movie);
            getSupportActionBar().setTitle(movie.getTitle());
        }

    }
}

File: app/src/main/java/com/example/movieslisttask/ui/activities/MovieInfoActivity.kt
package com.example.movieslisttask.ui.activities

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import com.example.movieslisttask.databinding.ActivityMovieInfoBinding
import com.example.movieslisttask.db.entities.Movie

/**
 * Created by alexkorolov on 2020-01-17.
 */
class MovieInfoActivity : AppCompatActivity() {

    lateinit var binding: ActivityMovieInfoBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Log.e("TAG", "onCreate: ")
        binding = ActivityMovieInfoBinding.inflate(layoutInflater)
        setContentView(binding.root)
        if (intent.hasExtra("movie")) {
            val movie: Movie = intent.getParcelableExtra("movie")
            binding.movie = movie
            supportActionBar?.title = movie.title
        }
    }
}